guides/release/tutorial/14-provider-components.md

```diff
@@ -0,0 +1,429 @@
+        owner: 'Veruca Salt',
+        city: 'San Francisco',
+        location: {
+          lat: 37.7749,
+          lng: -122.4194
+        },
+        category: 'Estate',
+        type: 'Standalone',
+        bedrooms: 15,
+        image: 'https://upload.wikimedia.org/wikipedia/commons/c/cb/Crane_estate_(5).jpg',
+        description: 'This grand old mansion sits on over 100 acres of rolling hills and dense redwood forests.'
+      },
+      {
+        id: 'urban-living',
+        title: 'Urban Living',
+        owner: 'Mike Teavee',
+        city: 'Seattle',
+        location: {
+          lat: 47.6062,
+          lng: -122.3321
+        },
+        category: 'Condo',
+        type: 'Community',
+        bedrooms: 1,
+        image: 'https://upload.wikimedia.org/wikipedia/commons/0/0e/Alfonso_13_Highrise_Tegucigalpa.jpg',
+        description: 'A commuters dream. This rental is within walking distance of 2 bus stops and the Metro.'
+      },
+      {
+        id: 'downtown-charm',
+        title: 'Downtown Charm',
+        owner: 'Violet Beauregarde',
+        city: 'Portland',
+        location: {
+          lat: 45.5175,
+          lng: -122.6801
+        },
+        category: 'Apartment',
+        type: 'Community',
+        bedrooms: 3,
+        image: 'https://upload.wikimedia.org/wikipedia/commons/f/f7/Wheeldon_Apartment_Building_-_Portland_Oregon.jpg',
+        description: 'Convenience is at your doorstep with this charming downtown rental. Great restaurants and active night life are within a few feet.'
+      }]
+    });
+
+    await render(hbs`<Rentals @rentals={{this.rentals}} />`);
+
+    assert.dom('.rentals').exists();
+    assert.dom('.rentals input').exists();
+
+    assert.dom('.rentals .results').exists();
+    assert.dom('.rentals .results li').exists({ count: 3 });
+
+    assert.dom('.rentals .results li:nth-of-type(1)').containsText('Grand Old Mansion');
+    assert.dom('.rentals .results li:nth-of-type(2)').containsText('Urban Living');
+    assert.dom('.rentals .results li:nth-of-type(3)').containsText('Downtown Charm');
+  });
+});
+```
+
+Now, if we try running our tests, they should all pass after making this change.
+
+<img src="/screenshots/14-provider-components/pass-1@2x.png" alt="The new test is passing." width="1024" height="1024">
+
+## Using Ember's `<Input>`
+
+Now that we have our component all set up, we can finally wire up our search box and store our search query! First things first: let's create a component class to store our query state.
+
+```js { data-filename="app/components/rentals.js" }
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export default class RentalsComponent extends Component {
+  @tracked query = '';
+}
+```
+
+Next, we'll wire up our query state in the component template.
+
+```handlebars { data-filename="app/components/rentals.hbs" data-diff="-4,+5" }
+<div class="rentals">
+  <label>
+    <span>Where would you like to stay?</span>
```

Comment:
I think we should show using explicit `<label for="some-id"> <input id="some-id">` format for accessibility. https://www.w3.org/WAI/tutorials/forms/labels/

Reply:
@jenweber @vaidehijoshi we don't have to do that because we nested the input inside the label, see https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-implicitly

Reply:
To elaborate, I agree, in a ideal world, all else being equal, it is _better_ to use explicit association. However, if it's done incorrectly, it'll do more harm than good. Specifically, if there are more than one element on the page using the same `id` attribute, things will be pretty broken.

So, it's quite risky to use `id` attributes in a component in general, since it is generally expected that you can invoke the same component multiple times on the same page, even though you may not have a need for them to be reusable as of the time of authoring the component. You can work around this by generating some sort of unique prefix/suffix to make the `id` unique, but that would be another non-trivial thing that we have to teach/explain, and the reader would have to really understand the caveat to be able to implement the technique correctly.

Given the risk, and given that implicit association is also an acceptable technique, and that the incompatibilities with implicit association are in browsers we don't support anyway (IE6), I think on balance, it is much better to teach the technique that is "plenty good enough" but very simple to understand/implement correctly, and go into more details about the alternative in the accessibility section.